src/test/java/com/google/sps/UserTripServletTest.java

```diff
@@ -0,0 +1,166 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.Config;
+import com.google.sps.data.User;
+import com.google.sps.servlets.AuthServlet;
+import com.google.sps.servlets.UserTripServlet;
+import com.google.sps.servlets.TripServlet;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(UserServiceFactory.class)
+public final class UserTripServletTest {
+
+  // Add constants necessary to testing retrieval of the current User.
+  public static final String EMAIL = "testemail@gmail.com";
+  public static final String AUTH_DOMAIN = "gmail.com";
+  public static final String LOGOUT_URL = "/_ah/logout?continue=%2F";
+  public static final String LOGIN_URL = "/_ah/login?continue=%2F";
```

Comment:
unless you use these outside they should be either private or protected!

Reply:
You're right, thanks -- fixed!